Assets/Scripts/PlayerIcon.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.UI;
+
+/// <summary>
+/// Manages Player Icon Location
+/// Appears when camera switch to map mode and when in isometric view
+/// 
+/// </summary>
+public class PlayerIcon : MonoBehaviour
+{
+    public Transform player;
+    private static bool _isoActive;
+    private static bool _mapActive;
+    private bool _onSlingShot;
+
+    private void Start()
+    {
+        
+        CameraSwitch.OnMapSwitch += SetMapActive;
+        IsometricStarView.OnIsometricStarView += SetIsometricActive;
```

Comment:
SetMapActive and SetIsometricActive are functionally identical, and can be condensed to a single method.